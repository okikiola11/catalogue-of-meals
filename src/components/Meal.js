import React from 'react';
import PropTypes from 'prop-types';

const Meal = ({ meal }) => {
  const { title, image } = meal;

  return (
    <div>
      <div className="image">My image</div>
      <div>{title}</div>
      <img src={image} alt={title} />
    </div>
  );
};

Meal.propTypes = {
  meal: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    image: PropTypes.string,
  }).isRequired,
};

export default Meal;
