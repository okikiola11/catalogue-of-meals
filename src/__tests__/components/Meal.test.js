import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';

import Meal from '../../components/Meal';

afterEach(cleanup);

const meal = {
  title: 'chicken & chips',
  image: 'image.jpg',
};

describe('Meal', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Meal meal={meal} />, div);
  });

  it('matches Meal snapshot', () => {
    const meals = renderer.create(<Meal meal={meal} />);
    const tree = meals.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders title correctly', () => {
    const { queryAllByTestId } = render(<Meal meal={meal} />);
    const title = queryAllByTestId('title');
    expect(title[0]).toBeTruthy();
  });

  it('renders image correctly', () => {
    const { queryAllByTestId } = render(<Meal meal={meal} />);
    const image = queryAllByTestId('image');
    expect(image[0]).toBeTruthy();
  });

  it('renders title correctly', () => {
    const { queryAllByTestId } = render(<Meal meal={meal} />);
    const title = queryAllByTestId('title');
    expect(title[0]).not.toBeFalsy();
  });

  it('renders image correctly', () => {
    const { queryAllByTestId } = render(<Meal meal={meal} />);
    const image = queryAllByTestId('image');
    expect(image[0]).not.toBeFalsy();
  });
});
