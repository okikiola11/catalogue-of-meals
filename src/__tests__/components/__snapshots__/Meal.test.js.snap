// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meal matches Meal snapshot 1`] = `
<div>
  <div
    className="meal-container"
  >
    <h3
      className="title"
      data-testid="title"
    >
      chicken & chips
    </h3>
    <div
      className="img-container"
    >
      <img
        alt="chicken & chips"
        data-testid="image"
        src="image.jpg"
      />
    </div>
  </div>
</div>
`;
